lib/proc-macros/src/content_hash.rs

```diff
@@ -45,6 +48,77 @@ pub fn generate_hash_impl(data: &Data) -> TokenStream {
                 quote! {}
             }
         },
-        _ => unimplemented!("ContentHash can only be derived for structs."),
+        // Generates a match statement with a match arm and hash implementation
+        // for each of the variants in the enum.
+        Data::Enum(ref data) => {
+            let match_hash_statements =
+                data.variants
+                    .iter()
+                    .enumerate()
+                    .map(|(i, v)| match &v.fields {
+                        Fields::Named(fields) => named_fields_match_arm(i, v, fields),
+                        Fields::Unnamed(fields) => anonymous_fields_match_arm(i, v, fields),
+                        Fields::Unit => {
+                            let ix = index_to_ordinal(i);
+                            quote_spanned! {v.span() =>
+                                Self::#v => {state.update(&#ix.to_le_bytes());}
```

Comment:
Maybe use the `::jj_lib::content_hash::ContentHash::hash(&#ix, state);` construction for consistency with the below cases? Shouldn't be a difference, but that would be more obvious if the code was the same.

Reply:
Thanks. I changed the other instances after we added an impl for u32 and forgot to update this one. Fixed.